data/maps/MP_Scenarios/Island_Hopping.wmf/scripting/hop_island.lua

```diff
@@ -14,9 +14,17 @@ function add_workers(hq, workers)
 end
 
 function add_soldiers(hq, soldiers)
-   local setpoints = {}
+   local setpoints = hq:get_soldiers("all")
    for sdescr, count in pairs(soldiers) do
-      setpoints[ {sdescr:match("(%d):(%d):(%d):(%d)")} ] = count
+      local desc = {sdescr:match("(%d):(%d):(%d):(%d)")}
```

Comment:
I tested this PR, but it doesn't fix the soldier problem for me. After some debugging, I've found that the problem is that `match()` returns strings, not numbers.

Reply:
well, it worked on my windows machine. will do some retest however.   
what was your result. did you have any soldiers on the second island at all. What were the numbers in the warehouse and what were it in the HQ on the 1st island?

Reply:
If I have 25 in the old HQ and 9 in the warehouse (1 goes to the new castle) then the new HQ gets 25.
Savegames before and after: [hopping.zip](https://github.com/widelands/widelands/files/10497844/hopping.zip)

Reply:
With the scenario in current master, the new HQ only gets the soldiers from the warehouse (9).

Reply:
Though my testcase differs from yours in that all soldiers are at the same level (0).

Reply:
found and fixed thanks for pointing me to use the correct key